File: crates/deslop/tests/fixtures/incremental-multilang/src/ledger_beta.py
# ledger_beta.py — the pasted copy of the Python reconciliation routine.


class BetaLedgerCursor:
    offset = 0


def reconcile_entries(entries, floor):
    balance = 0
    for entry in entries:
        if entry > floor:
            balance += entry * 2
        else:
            balance -= entry // 2
    return balance

File: crates/deslop/tests/fixtures/incremental-multilang/src/ledger_beta.ts
// ledger_beta.ts — the pasted copy of the TypeScript reconciliation routine.

export interface BetaLedgerCursor {
  offset: number;
}

export function reconcileEntries(entries: number[], floor: number): number {
  let balance = 0;
  for (const entry of entries) {
    if (entry > floor) {
      balance += entry * 2;
    } else {
      balance -= Math.trunc(entry / 2);
    }
  }
  return balance;
}
